### PR TITLE
Module to class

### PR DESCRIPTION
**Pull request by yelled3**
@seuros i'm fairly happy with this.

WDYT?

### CODE REVIEW

File: lib/sidekiq/active_record/task_worker.rb [hunk 1]
```diff
@@ -91,17 +90,37 @@ def identifier_key
...
+        # example:
+        #   sidekiq_task_model: AdminUser # or :admin_user
+        #
+        # then the worker will have access to `admin_user`, which is an alias to `task_model`
+        #
+        #   def perform_on_model
+        #     admin_user == task_model
+        #   end
+        #
+        def setup_task_model_alias(model_klass_name)
```
Comment (yelled3) [inline lib/sidekiq/active_record/task_worker.rb:104]: @seuros this is a one of the perks, for this approach.

File: lib/sidekiq/active_record/task_worker.rb [hunk 2]
```diff
@@ -58,29 +80,6 @@ def sidekiq_task_model(model_klass)
           get_sidekiq_task_options[:model_class] = model_klass
         end
 
-        def perform_on_model(model)
```
Comment (yelled3) [inline lib/sidekiq/active_record/task_worker.rb:61]: I also like the fact that we dont need to pass the model to every method...

File: lib/sidekiq/active_record/manager_worker.rb [hunk 3]
```diff
@@ -1,19 +1,14 @@
...
   module ActiveRecord
-    module ManagerWorker
-      extend Sidekiq::Worker
+    class ManagerWorker
+      include Sidekiq::Worker
 
       DEFAULT_IDENTIFIER_KEY = :id
       DEFAULT_BATCH_SIZE = 1000
 
-      def self.included(base)
```
Comment (yelled3) [inline lib/sidekiq/active_record/manager_worker.rb:9]: this mess is also not needed anymore

File: lib/sidekiq/active_record/manager_worker.rb [hunk 4]
```diff
@@ -1,19 +1,14 @@
 module Sidekiq
   module ActiveRecord
-    module ManagerWorker
-      extend Sidekiq::Worker
+    class ManagerWorker
+      include Sidekiq::Worker
 
       DEFAULT_IDENTIFIER_KEY = :id
       DEFAULT_BATCH_SIZE = 1000
```
Comment (seuros) [inline lib/sidekiq/active_record/manager_worker.rb:7]: we don't need that. default is 100.

File: lib/sidekiq/active_record/manager_worker.rb [hunk 5]
```diff
@@ -1,19 +1,14 @@
 module Sidekiq
   module ActiveRecord
-    module ManagerWorker
-      extend Sidekiq::Worker
+    class ManagerWorker
+      include Sidekiq::Worker
 
       DEFAULT_IDENTIFIER_KEY = :id
```
Comment (seuros) [inline lib/sidekiq/active_record/manager_worker.rb:6]: we don't need to use a constant. :id is always present in activerecord

Comment (yelled3) [inline lib/sidekiq/active_record/manager_worker.rb:6]: yeah, it's a `find_in_batches` thing:
if u dont add `:id`,  you get:

```
RuntimeError:
       Primary key not included in the custom select clause
```

https://github.com/rails/rails/blob/13d2696c10726afecd393753fcac88c5a9907d8c/activerecord/lib/active_record/relation/batches.rb#L123

File: lib/sidekiq/active_record/manager_worker.rb (hunk 4 above)
Comment (yelled3) [inline lib/sidekiq/active_record/manager_worker.rb:7]: default is 1000:
https://github.com/rails/rails/blob/13d2696c10726afecd393753fcac88c5a9907d8c/activerecord/lib/active_record/relation/batches.rb#L104

I want it to be clear for users, who are not familiar with find_in_batches

Comment (seuros) [inline lib/sidekiq/active_record/manager_worker.rb:7]: We can document that, no need to hardcode it again in this gem.

File: lib/sidekiq/active_record/manager_worker.rb (hunk 5 above)
Comment (seuros) [inline lib/sidekiq/active_record/manager_worker.rb:6]: we should be using pluck. there is no reason to build AR object.

Comment (yelled3) [inline lib/sidekiq/active_record/manager_worker.rb:6]: using `pluck` on a HUGE dataset is very costly. 
that's the whole point of the using `find_in_batches`

Comment (seuros) [inline lib/sidekiq/active_record/manager_worker.rb:6]: Let do some benchmark

Comment (seuros) [inline lib/sidekiq/active_record/manager_worker.rb:6]:
``` ruby
require 'active_support/core_ext/benchmark' 
50000.times {create(:user)} 
Benchmark.ms { User.pluck(:id).each {|id| puts id } }
#=> 5431.648426000001
Benchmark.ms { User.pluck(:id).each {|id| id } }
#=> 175.506035
```

Without index and sqlite3

Comment (yelled3) [inline lib/sidekiq/active_record/manager_worker.rb:6]: you're welcome to so, but I really don't see the need;
when dealing with BigData, it's just not feasible to pluck a few million model ids 
even aggregating them in-memory in ruby, seem costly :-)

Comment (yelled3) [inline lib/sidekiq/active_record/manager_worker.rb:6]: `50,000` is small.
what if we're talking about 10 million? 
also you some times, need to pluck other attributes (e.g user.status ) which can make things even worse

**Pull request merged**